lib/Conversion/SCFToCalyx/SCFToCalyx.cpp

```diff
@@ -1057,6 +1086,9 @@ LogicalResult BuildOpGroups::buildOp(PatternRewriter &rewriter,
       storeOp.getLoc(), memoryInterface.writeEn(),
       createConstant(storeOp.getLoc(), rewriter, *getComponent(), 1, 1));
   rewriter.create<calyx::GroupDoneOp>(storeOp.getLoc(), memoryInterface.done());
+
+  getComponentState().registerNonPipelineOperations(storeOp, group);
```

Comment:
*sigh* I was hoping to avoid more pipeline specific state until some of this is reworked. However, given the current phase ordering of how this pass does partial lowering, I think this is a reasonable fix for the short term. I haven't really thought about if there is a better way to solve this.

Reply:
Yes I think that is the issue as well. There are a lot of inter-dependencies between the two lowering processes, which was nice initially but now is showing its limitations. I think finding a more general solution will become obvious when these are separated.

Reply:
Yeah, I don't think I will honestly have the cycles to come up with a better approach in the short term, or if there even is one given the current phase ordering. This makes sense to move forward with.